prisma/schema.prisma

```diff
@@ -585,16 +590,16 @@ model ModelFile {
 }
 
 model File {
-  id        Int      @id @default(autoincrement())
-  name      String
-  url       String
-  sizeKB    Float
-  createdAt DateTime @default(now())
-  articleId Int?
-  article   Article? @relation(fields: [articleId], references: [id], onDelete: Cascade)
-  metadata  Json?
+  id         Int      @id @default(autoincrement())
+  name       String
+  url        String
+  sizeKB     Float
+  createdAt  DateTime @default(now())
+  entityId   Int
+  entityType String
+  metadata   Json?
 
-  @@index([articleId], type: Hash)
+  @@unique([id, entityId, entityType])
```

Comment:
This unique index doesn't offer benefits as id is already unique and it doesn't allow fast lookups for entityId / entityType. If the intent is to search by [entityType, entityId] then we should just have an index on that combo (in that order). This would require queries to always include the entityType when searching for an entityId. If this is not feasible then I recommend just putting an index on entityId as that would offer both speed and flexibility, This index can be unique if you only ever allow 1 file per entity.